### DISCUSSION TITLE
Can you make slangc emit a `#[shader_stage]` when compiling to glsl?

### DISCUSSION THREAD

**Discussion by LouChiSoft** [shader-slang/slang#4433] [Q&A] [answered]
Hi, I want to evaluate what it's like using Slang in Godot. Godot by default uses GLSL for compute shaders, but it doesn't recognise the GLSL files as a valid shader without a corresponding `#[compute]` tag at the beginning of the file. It's not the end of the world to have to add it manually, but if you are iterating on a shader it could get very repetitive and time consuming so was wondering if there was a way to force slangc to do that for me.

Thanks

**Comment by csyonghe** [comment]
You can use the requirePrelude feature to do that.

Check out https://shader-slang.com/slang/user-guide/a1-04-interop.html

**Reply by LouChiSoft** [reply]
Thanks, I had a quick try but I didn't seem to be able to get it to work. I've tried the prelude at both the beggining of the shader, but that's not where that macro is meant to be used according to the LSP. And when trying to compile it at the beginning of the main function it returns an `unimplemented feature in slang compiler` error, but I am using the latest version. Does it only work for CPU code (the example in the link looked to only be using C++)

**Reply by csyonghe** [reply]
Can you provide your code and command line so we can take a look?

**Reply by LouChiSoft** [reply]
Sure. It's just the basic user guide example:

```
__requirePrelude(R"(#[compute])"); //<-- attempt 1: "unexpected string literal"
StructuredBuffer<float> buffer0;
StructuredBuffer<float> buffer1;
RWStructuredBuffer<float> result;

[shader("compute")]
[numthreads(1,1,1)]
void computeMain(uint3 threadId: SV_DispatchThreadID)
{
    __requirePrelude(R"(#[compute])"); // <-- attempt 2: "unimplemented feature"
    uint index = threadId.x;
    result[index] = buffer0[index] + buffer1[index];
}
```

And compile command `slangc compute.slang -profile glsl_450 -target glsl -o compute.glsl -entry computeMain`

**Reply by csyonghe** [reply, marked as answer]
Only the second attempt is supported. It seems the problem is that prelude isn't supported by the glsl backend, which should be easy to add.

**Reply by LouChiSoft** [reply]
Ah, fair enough. Manually editing for me until it's added then. Thanks
